ft-buyer-signup #3

patsicko opened the pull request:
# What this PR do
# Create a buyer signup endpoint
# Description of the task
- Create user model
- Create buyer controller
- Create buyer api endpoint
- Test buyer api with jest
- Document buyer api with swagger

## How should this be manually tested

- clone the repo
- use npm install to install all packages
- Add all required environment variables following the .env.example provided
- use npx sequelize db:migrate to migrate migrations
- use npm run dev to start the server
use the appropriate router in postman or thunderclient 
navigate to  the http://localhost:PORT  by replacing 'PORT' the port that your the app is listening to, and test it using the documented API. By sending a request to create a buyer, the response will be displayed containging a newly created buyer and welcome message will be sent to the provided email
- Run npm test to test the app

## Trello story id
#003

sostene12 reviewed (commented):
you can add more details on this pull, like profile model and work hand in hand with task of aimable

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

sostene12 reviewed (commented):
- please solve the conflicts

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

Shumba250 committed:
Merge branch 'develop' into ft-vendor-register

Commit:
ft-buyer-signup
-
- Change config file buy adding test database
 - updated .env file
[Finishes]

Commit:
ft-buyer-signup
- Create uuser model
- Create buyer controller
- Create buyer post endpoint
- Create api documentation with swagger
- Test api with jest
- Add cross-env package
- add migrate:up:test & migrate:undo:test
[Finishes#3]

Commit:
ft-buyer-signup
- Add test database variables in .env.example
[Finishes#3]

Commit:
ft-buyer-signup
 - Add test case scenario for wrong email
[Finishes#3]

Commit:
ft-buyer-signup
 - Change Test database variables in ci.yml
[Finishes#3]

Commit:
ft-buyer-signup
 - Change Test database variables in ci.yml
[Finishes#3]

Commit:
ft-buyer-signup
- Implement the sent of emil to a successfully registered buyer
[Finishes#]

wakaflorien reviewed (changes requested):
- **TEST_DB_HOST, TEST_DB_USERNAME, TEST_DB_PASSWORD** are not needed.
- existing on conflict error codes should be 409
- no need to push console logs in your codes once they served your purpose
- profile model still unnormalized e.g how is **businessNumber** attribute relevant to **buyer**. same goes to **businessAddress**, **taxIdNumber, typeOfProducts**
- the PR description doesn't well guide to how this can be manually tested

wakaflorien commented on src/controllers/buyerController.js:
```diff
@@ -0,0 +1,73 @@
+console.log(buyer);
+    buyer.save();
+
+    const token = jwt.sign({ id: buyer.id }, process.env.JWT_SECRET, {
```
- when the JWT_SECRET is no found the account is still created.

Commit:
ft-buyer-signup
- Remove console.log()
[Finishes]

Commit:
ft-buyer-signup
- Create buyer model
- Create relatipon of buyer model to user model
- Create buyers migration
[Finishes#]

wakaflorien commented on test/user.test.js:
```diff
@@ -8,14 +8,47 @@ describe("testing all routes", () => {
+    
+    });
+
+ console.log(response.body);
```
- no need of leaving this console logs @patsicko
